pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/index/nullvalue/NullValueIndexType.java

```diff
@@ -77,11 +80,23 @@ public String getPrettyName() {
 
   @Override
   public ColumnConfigDeserializer<IndexConfig> createDeserializer() {
-    return IndexConfigDeserializer.ifIndexingConfig(
-        IndexConfigDeserializer.alwaysCall((TableConfig tableConfig, Schema schema) ->
-            tableConfig.getIndexingConfig().isNullHandlingEnabled()
-                ? IndexConfig.ENABLED
-                : IndexConfig.DISABLED));
+    return (TableConfig tableConfig, Schema schema) -> {
+      Collection<FieldSpec> allFieldSpecs = schema.getAllFieldSpecs();
+      Map<String, IndexConfig> configMap = Maps.newHashMapWithExpectedSize(allFieldSpecs.size());
+
+      boolean enableColumnBasedNullHandling = schema.isEnableColumnBasedNullHandling();
+
+      for (FieldSpec fieldSpec : allFieldSpecs) {
+        IndexConfig indexConfig;
+        if (!enableColumnBasedNullHandling || fieldSpec.isNullable()) {
```

Comment:
For my understanding, is this to keep the old behavior of always loading the null vector when the index exists?
In the existing code, we do check `tableConfig.getIndexingConfig().isNullHandlingEnabled()` but the check is dropped here. Is that because we do not check whether it is enabled in the existing code?

Reply:
You are right, although in my understanding nobody actually used this to decide whether the index is created/consumed or not. Even then, I'll change this code to be sure it is still backward compatible

Reply:
Fixed in 9402a7c4a65a9d031f66a696031bc4154d217915. I think it is also easier to understand now.